src/gui/processing/qgsprocessingwidgetwrapperimpl.cpp

```diff
@@ -407,6 +409,82 @@ QgsAbstractProcessingParameterWidgetWrapper *QgsProcessingStringWidgetWrapper::c
 
 
 
+//
+// QgsProcessingAuthConfigWidgetWrapper
+//
+
+QgsProcessingAuthConfigWidgetWrapper::QgsProcessingAuthConfigWidgetWrapper( const QgsProcessingParameterDefinition *parameter, QgsProcessingGui::WidgetType type, QWidget *parent )
+  : QgsAbstractProcessingParameterWidgetWrapper( parameter, type, parent )
+{
+
+}
+
+QWidget *QgsProcessingAuthConfigWidgetWrapper::createWidget()
+{
+  switch ( type() )
+  {
+    case QgsProcessingGui::Standard:
+    case QgsProcessingGui::Modeler:
+    case QgsProcessingGui::Batch:
+    {
+      mAuthConfigSelect = new QgsAuthConfigSelect();
+      mAuthConfigSelect->setToolTip( parameterDefinition()->toolTip() );
+
+      connect( mAuthConfigSelect, &QgsAuthConfigSelect::selectedConfigIdChanged, this, [ = ]
+      {
+        emit widgetValueHasChanged( this );
+      } );
+      return mAuthConfigSelect;
+    };
+  }
+  return nullptr;
+}
+
+void QgsProcessingAuthConfigWidgetWrapper::setWidgetValue( const QVariant &value, QgsProcessingContext &context )
+{
+  const QString v = QgsProcessingParameters::parameterAsString( parameterDefinition(), value, context );
+  if ( mAuthConfigSelect )
+    mAuthConfigSelect->setConfigId( v );
+}
+
+QVariant QgsProcessingAuthConfigWidgetWrapper::widgetValue() const
+{
+  if ( mAuthConfigSelect )
+    return mAuthConfigSelect->configId();
+  else
+    return QVariant();
+}
+
+QStringList QgsProcessingAuthConfigWidgetWrapper::compatibleParameterTypes() const
+{
+  return QStringList()
+         << QgsProcessingParameterAuthConfig::typeName()
+         << QgsProcessingParameterString::typeName()
+         << QgsProcessingParameterExpression::typeName();
+}
+
+QStringList QgsProcessingAuthConfigWidgetWrapper::compatibleOutputTypes() const
+{
+  return QStringList() << QgsProcessingOutputNumber::typeName()
+         << QgsProcessingOutputFile::typeName()
+         << QgsProcessingOutputString::typeName();
```

Comment:
should be only QgsProcessingOutputString::typeName() ?

Reply:
Oops - good point! I can't see any valid use case for someone using an output file name as a authentication configuration ID :yum: